Useful_Links.rst: Adds new page

CosminIcatoiu opened the pull request:
Especially for newcomers there was an issue about
their orientation through the site in terms of finding
specific materials that can help them. Adds a new page
with the most useful stuff from a newcomer`s perspective.
Edits the index file from the root folder so that the
document can be added to the page.

Closes https://github.com/coala/coala/issues/3515

<!--
Thanks for your contribution!

Reviewing pull requests takes a lot of time and we're all volunteers. Please make sure you go through the following checklist and all items before pinging someone for a review.
-->

### Checklist

- [x] I have rebased properly. Please see [our tutorial on rebasing](http://coala.io/git#rebasing).
- [x] I have gone through the [commit guidelines](http://coala.io/commit) and I've followed them.
- [x] I have followed the [guidelines for the commit shortlog](http://coala.io/commit#shortlog).
- [x] I have followed the [guidelines for the commit body](http://coala.io/commit#commit-body).
- [x] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://coala.io/commit#issue-reference).
- [x] I have [run all the tests](http://api.coala.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### Reviewers

<!--
Please list the Github handles of people you think susceptible to review this PR. Feel free to leave this section blank if you don't know who to tag.
-->

<!--
End note:

As you learn things over your Pull Request please help others on the chat and on PRs to get their stuff right as well!
-->

Mixih commented on docs/Developers/Useful_Links.rst:
```diff
@@ -0,0 +1,47 @@
+
+.. note::
+
+    **If you ever feel that during your learning or developping**
```
`developping` typo

Mixih commented on docs/Developers/Useful_Links.rst:
```diff
@@ -0,0 +1,47 @@
+information.
+
+.. note::
+
```
why bold?

Mariatta replied:
Right, don't make this bold. Too obtrusive. This will already be rendered in a box.

Mariatta replied:
I'd really like to reduce the amount of boxes we use in our docs. They're becoming too distracting.
This can be just a paragraph on its own, for example:
```
Feel free to suggest a new links to be added here, by creating an issue in our issue tracker.
```

CosminIcatoiu replied:
I feel it`s important to be bold or in a box so that people would see it easier and notice that they can contribute. I feel that a separate paragraph is easier to be ignored in this case. ( you have just links, not a full doc ) I agree that using both is unnecessary

Mixih commented on docs/Developers/Useful_Links.rst:
```diff
@@ -0,0 +1,47 @@
+
+The purpose of this document is to gather links that coala
+developers usually use throughout their work and therefore
+make it easier for everybody to find and access that specific
```
I feel `specific` is unnecessary.

Mariatta replied:
Just `The purpose of this document is to gather links that coala developers usually use throughout their work.`
The rest is not necessary.

Mixih commented:
also in commit shortlog: adds -> add. (adds is not imperative mood)

Mariatta commented on docs/Developers/Useful_Links.rst:
```diff
@@ -0,0 +1,47 @@
+Git-Links
+---------
+
+`Git Tutorial <https://try.github.io/levels/1/challenges/1>`_
```
Maybe these should be bulletted list items?

CosminIcatoiu replied:
OK :)

Mariatta commented on docs/Developers/Useful_Links.rst:
```diff
@@ -0,0 +1,47 @@
+------------
+
+`Code Style <https://www.python.org/dev/peps/pep-0008/>`_
+`Python Tutorial <https://docs.python.org/2/tutorial/>`_
```
Link to python 3 tutorial

CosminIcatoiu commented:
Tell me your opinion now :) Think i addressed all the points ( not very sure that in the proper way though)

Mariatta commented on docs/Developers/Useful_Links.rst:
```diff
@@ -0,0 +1,42 @@
+developers usually use throughout their work.
+.. note::
+
+    If you ever encounter a link that helped you or that is not
```
I still don't think this portion warrants a box.

The Notes box is visually intrusive, and should be reserved for something of utmost importance.

Suggesting people to submit a new link is just a suggestion, and not absolutely necessary.
If they miss reading this particular paragraph the first time, it's fine for me. 
The most important part of this document is the list of links below.

CosminIcatoiu commented:
Ok . I will change it when i we'll be back home.

CosminIcatoiu commented:
Don`t really know how to squash the commits . When i apply git rebase -i it appears noop, not 3 commits and at every commit i add all the 3 files and then commit

Mariatta commented on docs/Developers/Newcomers_Guide.rst:
```diff
@@ -345,5 +345,7 @@ go for `a low issue <https://coala.io/low>`__ - they are really rewarding!
     you!
 
+    If you ever have problems in finding some links maybe you can find
+    the solution in our :doc:`<useful links section <Useful_Links>`.
```
remove the < before useful

Mariatta commented on docs/Developers/Useful_Links.rst:
```diff
@@ -0,0 +1,41 @@
+---------
+
+-   `Git Tutorial <https://try.github.io/levels/1/challenges/1>`_
+-   `Coala's Git Tutorial <http://api.coala.io/en/latest/Developers/Git_Basics.html>`_
```
Use internal link
eg 
<Git_Basics>

Mariatta commented on docs/Developers/Useful_Links.rst:
```diff
@@ -0,0 +1,41 @@
+-   `Git Tutorial <https://try.github.io/levels/1/challenges/1>`_
+-   `Coala's Git Tutorial <http://api.coala.io/en/latest/Developers/Git_Basics.html>`_
+-   `Git Commits <https://wiki.gnome.org/Git/CommitMessages>`_
+-   `Coala Commits <http://api.coala.io/en/latest/Developers/Writing_Good_Commits.html>`_
```
Use internal link

Mariatta commented:
Thanks @CosminIcatoiu .
I have a few more comments. But otherwise look good :)

CosminIcatoiu commented:
@Mariatta have a look at the changes I did when you have time :). Thanks

CosminIcatoiu commented:
Should i do anything apart from rebasing it ?

Mariatta commented:
ack f920799

Mariatta commented:
@rultor merge

rultor commented:
> @rultor merge

@mariatta OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/10610-272545782)

rultor commented:
> @rultor merge

@cosminicatoiu @mariatta Oops, I failed. You can see the full log [here](http://www.rultor.com/t/10610-272545782) (spent 1min)

```
 * [new branch]      sils/log   -> fork/sils/log
 * [new branch]      sils/loga  -> fork/sils/loga
 * [new branch]      sils/loggi -> fork/sils/loggi
 * [new branch]      sils/minlog -> fork/sils/minlog
 * [new branch]      sils/newc  -> fork/sils/newc
 * [new branch]      sils/nextgen2 -> fork/sils/nextgen2
 * [new branch]      sils/perf  -> fork/sils/perf
 * [new branch]      sils/proto -> fork/sils/proto
 * [new branch]      sils/quotes -> fork/sils/quotes
 * [new branch]      sils/resf  -> fork/sils/resf
 * [new branch]      sils/resoverlap -> fork/sils/resoverlap
 * [new branch]      sils/resultfilter -> fork/sils/resultfilter
 * [new branch]      sils/reviewing -> fork/sils/reviewing
 * [new branch]      sils/sem   -> fork/sils/sem
 * [new branch]      sils/setup -> fork/sils/setup
 * [new branch]      sils/swtest -> fork/sils/swtest
 * [new branch]      sils/trimlin -> fork/sils/trimlin
 * [new branch]      sils/trumpscript -> fork/sils/trumpscript
 * [new branch]      sils/tut   -> fork/sils/tut
 * [new branch]      sils/typeer -> fork/sils/typeer
 * [new branch]      sils/win   -> fork/sils/win
 * [new branch]      sils/wip/cachingprop -> fork/sils/wip/cachingprop
 * [new branch]      sils/wippatch -> fork/sils/wippatch
 * [new branch]      sils/xml   -> fork/sils/xml
 * [new branch]      tulphoon/uselog -> fork/tulphoon/uselog
 * [new branch]      ud/SecInherit -> fork/ud/SecInherit
 * [new branch]      ud/spa-ci  -> fork/ud/spa-ci
 * [new branch]      udayan/fileproxy -> fork/udayan/fileproxy
 * [new branch]      udayan/newbears -> fork/udayan/newbears
 * [new branch]      underyx/req-version-specs -> fork/underyx/req-version-specs
 * [new branch]      wip/sils/profiling -> fork/wip/sils/profiling
 * [new branch]      wip/udayan/filesim -> fork/wip/udayan/filesim
 * [new branch]      wip/udayan/gui2 -> fork/wip/udayan/gui2
 * [new branch]      wip/udayan/guiproto -> fork/wip/udayan/guiproto
 * [new branch]      zimmermann/collecting -> fork/zimmermann/collecting
 * [new branch]      zimmermann/config -> fork/zimmermann/config
 * [new branch]      zimmermann/debug -> fork/zimmermann/debug
 * [new branch]      zimmermann/languages -> fork/zimmermann/languages
+ args=
+ '[' only == default ']'
+ '[' only == no ']'
+ '[' only == only ']'
+ args=' --ff-only'
+ export BRANCH=__rultor
+ BRANCH=__rultor
++ git show-branch __rultor
++ wc -l
+ '[' 0 -gt 0 ']'
+ git checkout -B __rultor fork/cosmin_links
Switched to a new branch '__rultor'
Branch __rultor set up to track remote branch cosmin_links from fork.
+ git checkout -B master origin/master
Switched to and reset branch 'master'
Branch master set up to track remote branch master from origin.
+ '[' false == true ']'
+ '[' false == true ']'
+ git merge --ff-only __rultor
fatal: Not possible to fast-forward, aborting.
'cid' file is absent, container wasn't started correctly

```

Adrianzatreanu commented on docs/Developers/Useful_Links.rst:
```diff
@@ -0,0 +1,41 @@
+
+If you ever encounter a link that helped you or that is not
+a part of the document and should be, feel free to suggest it
+by creating an issue in our issue tracker.
```
maybe give a link to opening an issue in this repo or so?

Adrianzatreanu commented on docs/Developers/Useful_Links.rst:
```diff
@@ -0,0 +1,41 @@
+-----------
+
+-   `Install coala <http://docs.coala.io/en/latest/Users/Install.html>`_
+-   `Coala Issues <https://github.com/coala/coala/issues>`_
```
never write coala with uppercase C, you're triggering @sils

Adrianzatreanu commented on docs/Developers/Useful_Links.rst:
```diff
@@ -0,0 +1,41 @@
+---------
+
+-   `Git Tutorial <https://try.github.io/levels/1/challenges/1>`_
+-   :doc:`Coala's Git Tutorial <Git_Basics>`
```
Coala -> coala

Adrianzatreanu commented on docs/Developers/Useful_Links.rst:
```diff
@@ -0,0 +1,41 @@
+Git-Links
+---------
+
+-   `Git Tutorial <https://try.github.io/levels/1/challenges/1>`_
```
i feel like these hyperlinks can have their descriptions made better. for example: Git Commits -> Commit message guidelines
Git rebase -> How to rebase

Adrianzatreanu commented on docs/Developers/Useful_Links.rst:
```diff
@@ -0,0 +1,41 @@
+Python-Links
+------------
+
+-   `Code Style <https://www.python.org/dev/peps/pep-0008/>`_
```
feel like this could be added somewhere too https://api.coala.io/en/latest/Developers/Codestyle.html

sils replied:
I'd remove pep 8 in favour of the coala codestyle link

sils replied:
also this should go into the coala links section

Adrianzatreanu commented on docs/Developers/Useful_Links.rst:
```diff
@@ -0,0 +1,41 @@
+coala-Links
+-----------
+
+-   `Install coala <http://docs.coala.io/en/latest/Users/Install.html>`_
```
you should also include http://api.coala.io/en/latest/Developers/Review.html as this points on how the review process is done

Adrianzatreanu commented:
wont merge, needs a rebase. however, i reviewed it a little too so should be changed a bit :)

Mariatta commented:
Thanks @Adrianzatreanu . Sorry, I missed those earlier about lowercase c :(
unack f920799

Mariatta commented:
unack f920799

CosminIcatoiu commented:
@Adrianzatreanu @Mariatta i think i changed everything Adrian said. Have a look once again :) . I also rebased it so that it can be merged. Lastly please check if i addressed well links from the same folder ( the ones with :doc: ) When i checked the whole file didn't really looked as i thought but i am not very experienced :) Thanks again to both of you for advice.

CosminIcatoiu commented:
@Mariatta can you check it again and tell me if it is ready for rebase? and then merge?

sils commented on docs/Developers/Useful_Links.rst:
```diff
@@ -0,0 +1,45 @@
+-   `Syntax <http://www.sphinx-doc.org/en/stable/markup/toctree.html#toctree-directive>`_
+
+coala-Links
+-----------
```
I'd first and foremost include coala.io/links here which is an ever updating collection of the coala shortlinks!

sils commented on docs/Developers/Useful_Links.rst:
```diff
@@ -0,0 +1,45 @@
+-   `How to rebase <https://asciinema.org/a/78683>`_
+-   `Rebase Concept <http://jeffkreeftmeijer.com/2010/the-magical-and-not-harmful-rebase/>`_
+-   `Short Rebase Tutorial <http://stackoverflow.com/questions/7244321/how-do-i-update-a-github-forked-repository>`_
+-   `coala Git Repository <https://github.com/coala/>`_
```
this isn't even the repo this is the org

sils commented:
shortlog: how about `Add Useful_Links.rst` and drop the context?

CosminIcatoiu commented:
I will change what you said but today i had no network si couldn't commit anything

sils commented on docs/Developers/Useful_Links.rst:
```diff
@@ -0,0 +1,45 @@
+by creating an issue in our `issue tracker <https://github.com/coala/coala/issues/new>`_.
+
+Git-Links
+---------
```
maybe we should have sub headings like general tutorials, rebase, important repos or so

CosminIcatoiu replied:
Don't really know... i feel there are not enough links now for sub headings ... maybe in the future. What do you think. Still no network ....

CosminIcatoiu commented:
@sils I think i changed what you said. have a check. Still, i believe that the subheadings are not needed yet. Most of them would have only one link. Am i wrong?

sils commented:
this is ok for me.

CosminIcatoiu commented:
I will rebase it for merge :) tonight

sils commented:
reack https://github.com/coala/coala/pull/3539/commits/20b938527210360373233550092cf928f24f3f93

sils commented:
@rultor merge

rultor commented:
> @rultor merge

@sils OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/10610-274320315)

rultor commented:
> @rultor merge

@sils Done! FYI, the full log is [here](http://www.rultor.com/t/10610-274320315) (took me 1min)
